python/cudf/dataframe/dataframe.py

```diff
@@ -895,10 +907,13 @@ def _prepare_series_for_add(self, col, forceindex=False):
         series = self._sanitize_values(series, SCALAR)
 
         empty_index = len(self._index) == 0
-        if forceindex or empty_index or self._index.equals(series.index):
```

Comment:
You need this check still otherwise the series won't have the same index as the dataframe. The last condition should just be changed to `self.index is series.index` to check they both reference the same index object.